lib/modules/datasource/nuget/v3.ts

```diff
@@ -233,4 +258,54 @@ export class NugetV3Api {
 
     return dep;
   }
+
+  @cache({
+    namespace: NugetV3Api.cacheNamespace,
+    key: (
+      _http: Http,
+      registryUrl: string,
+      packageName: string,
+      _packageVersion: string | null,
+      _nupkgUrl: string,
+    ) => `source-url:${registryUrl}:${packageName}`,
+    ttlMinutes: 10080, // 1 week
+  })
+  async getSourceUrlFromNupkg(
+    http: Http,
+    _registryUrl: string,
+    packageName: string,
+    packageVersion: string | null,
+    nupkgUrl: string,
+  ): Promise<string | null> {
+    // istanbul ignore if: experimental feature
+    if (!process.env.RENOVATE_X_NUGET_DOWNLOAD_NUPKGS) {
+      logger.debug(
+        `Skipping nupkg download because RENOVATE_X_NUGET_DOWNLOAD_NUPKGS is not set.`,
+      );
```

Comment:
Won't this log this debug message once per package for every user not using the option?

Reply:
Yes, good point. `logger.once`?

Reply:
In its current state I think it's at risk of alarming people / creating false concern. If you're certain this should be debug then keeping it to `RENOVATE_X_NUGET_DOWNLOAD_NUPKGS is not set` is better so that the majority will just ignore it

Reply:
Adapted ✅